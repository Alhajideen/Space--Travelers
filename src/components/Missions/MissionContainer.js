import React, { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import '../../styles/mission.css';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions, joinMission } from '../../redux/Mission/missionSlice';

function MissionContainer() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.mission.data);

  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

  const handleMember = (id) => {
    dispatch(joinMission(id));
  };
  return (
    <div className="mission-container my-3">
      <Table striped bordered hover size="sm">
        <thead>
          <tr>
            <th>Missions</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => {
            if (!mission.member) {
              return (
                <tr key={mission.mission_id} className="table-rows">
                  <td className="mission-name">{mission.mission_name}</td>
                  <td>{mission.description}</td>
                  <td className="status">
                    <button type="button" className="not-member p-1">
                      Not a Member
                    </button>
                  </td>
                  <td className="status">
                    <button
                      type="button"
                      className="join rounded-1 p-1"
                      onClick={() => handleMember(mission.mission_id)}
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
              );
            }
            return (
              <tr key={mission.mission_id} className="table-rows">
                <td className="mission-name">{mission.mission_name}</td>
                <td>{mission.description}</td>
                <td className="status">
                  <button type="button" className="member p-1">
                    Active Member
                  </button>
                </td>
                <td className="status">
                  <button
                    type="button"
                    className="leave p-1"
                    onClick={() => handleMember(mission.mission_id)}
                  >
                    Leave Mission
                  </button>
                </td>
              </tr>
            );
          })}
        </tbody>
      </Table>
    </div>
  );
}

export default MissionContainer;
