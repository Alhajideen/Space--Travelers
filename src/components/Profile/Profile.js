import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';

function Profile() {
  const missions = useSelector((state) => state.mission.data);
  const [mission, setMission] = useState([]);
  const [empty, setEmpty] = useState(false);
  useEffect(() => {
    const joinedMisssion = missions.filter((item) => item.member === true);
    setMission(joinedMisssion);
    if (joinedMisssion.length === 0) {
      setEmpty(true);
    }
  }, []);
  return (
    <div>
      <div className="my-missions">
        <div className="header-text">
          <h1>My Missions</h1>
        </div>
        <table>
          <tbody>
            {!empty
              && mission.map((item) => (
                <tr key={item.mission_id}>
                  <td>{item.mission_name}</td>
                </tr>
              ))}
            {empty && (
              <tr>
                <td>You have not joined any mission.</td>
              </tr>
            )}
          </tbody>
        </table>
      </div>
    </div>
  );
}

export default Profile;
