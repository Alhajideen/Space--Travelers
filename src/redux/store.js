import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import missionReducer from './Mission/missionSlice';
import rocketReducer from './Rockets/rocketSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    rockets: rocketReducer,
    mission: missionReducer,
  },
});
