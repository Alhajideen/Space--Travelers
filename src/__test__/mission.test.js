import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import MissionContainer from '../components/Missions/MissionContainer';
import Profile from '../components/Profile/Profile';
import '@testing-library/jest-dom';

describe('Testing Profile and Mission Components', () => {
  it('Testing MissionContainer components', () => {
    const myMission = render(
      <React.StrictMode>
        <Router>
          <Provider store={store}>
            <MissionContainer />
          </Provider>
        </Router>
      </React.StrictMode>,
    );
    expect(myMission).toMatchSnapshot();
  });

  it('Testing MissionContainer components', () => {
    const profile = render(
      <React.StrictMode>
        <Router>
          <Provider store={store}>
            <Profile />
          </Provider>
        </Router>
      </React.StrictMode>,
    );
    expect(profile).toMatchSnapshot();
  });
});
