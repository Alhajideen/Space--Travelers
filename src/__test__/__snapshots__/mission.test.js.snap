// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Profile and Mission Components Testing MissionContainer components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mission-container my-3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Missions
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
              <th>
                 
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mission-container my-3"
    >
      <table
        class="table table-sm table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Missions
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
            <th>
               
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Profile and Mission Components Testing MissionContainer components 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-container"
      >
        <div
          class="my-missions"
        >
          <div
            class="header-text"
          >
            <h2>
              My Missions
            </h2>
          </div>
          <table
            class="mission-table"
          >
            <tbody
              class="mission-tbody"
            >
              <tr>
                <td>
                  You have not joined any mission.
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="my-rockets"
        >
          <div
            class="header-text"
          >
            <h2>
              My Rockets
            </h2>
          </div>
          <table
            class="rockets-table"
          >
            <tbody>
              <tr>
                <td>
                  You have not reserved any rocket.
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-container"
    >
      <div
        class="my-missions"
      >
        <div
          class="header-text"
        >
          <h2>
            My Missions
          </h2>
        </div>
        <table
          class="mission-table"
        >
          <tbody
            class="mission-tbody"
          >
            <tr>
              <td>
                You have not joined any mission.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="my-rockets"
      >
        <div
          class="header-text"
        >
          <h2>
            My Rockets
          </h2>
        </div>
        <table
          class="rockets-table"
        >
          <tbody>
            <tr>
              <td>
                You have not reserved any rocket.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
